src/extension.ts

```diff
@@ -710,7 +710,10 @@ export async function activate(context: vscode.ExtensionContext) {
                 CONSTANTS.ERROR.NO_FILE_TO_RUN,
                 DialogResponses.MESSAGE_UNDERSTOOD
             );
-        } else if (!utils.validCodeFileName(currentFileAbsPath)) {
+        } else if (
+            device == CONSTANTS.DEVICE_NAME.CPX &&
+            !utils.validCodeFileName(currentFileAbsPath)
```

Comment:
Have you thought about renaming the file in the background and deploying it? That we way we won't have to bother the user to rename it.

Reply:
Rename the file in the background then rename it back after it is done?

Reply:
I was thinking more of how in unix you can do copy and rename at the same time:
`$ cp local_disk/python_script.py microbit/code.py`
But whatever you think makes sense should be good. Doesn't have to be in this PR.

Reply:
I see what you mean now. I'll update it in this PR :D

Reply:
Sounds good! Do you think there could be some drawbacks to this approach (taking care of the script name in the background rather than explicitly asking the user to rename the file)?

Reply:
There could be like if someone makes a file 'main.py', deploys it and then makes a file called 'code.py' and then deploys it, there may be problems because it may still be running the wrong file so we could just force it to be named 'code.py' always.

In response to the OG question, I think the pros outweigh the cons especially because I can remove a lot of unnecessary pop ups now.